apps/src/lib/levelbuilder/lesson-editor/AddLevelDialog.jsx

```diff
@@ -130,38 +165,54 @@ export default class AddLevelDialog extends Component {
-              <button type="button" value={'Create New Level'}>
-                Create New Level
-              </button>
-            </ToggleGroup>
-            {this.state.methodOfAddingLevel === 'Find Level' && (
-              <div style={styles.filtersAndLevels}>
-                <AddLevelFilters
-                  searchFields={this.state.searchFields}
-                  handleSearch={this.handleSearch}
-                />
-                <AddLevelTable
+          {this.state.levels && this.state.searchFields && (
+            <div style={styles.topArea}>
+              <ToggleGroup
+                selected={this.state.methodOfAddingLevel}
+                onChange={this.handleToggle}
+              >
+                <button type="button" value={'Find Level'}>
+                  Find Level
+                </button>
+                <button type="button" value={'Create New Level'}>
+                  Create New Level
+                </button>
+              </ToggleGroup>
+              {this.state.methodOfAddingLevel === 'Find Level' && (
+                <div style={styles.filtersAndLevels}>
+                  <AddLevelFilters
+                    searchFields={this.state.searchFields}
+                    handleSearch={this.handleNewSearch}
+                    handleChangeLevelName={this.handleChangeLevelName}
+                    handleChangeLevelType={this.handleChangeLevelType}
+                    handleChangeScript={this.handleChangeScript}
+                    handleChangeOwner={this.handleChangeOwner}
+                    ownerId={this.state.ownerId}
+                    scriptId={this.state.scriptId}
+                    levelName={this.state.levelName}
+                    levelType={this.state.levelType}
+                  />
+                  <AddLevelTable
+                    setCurrentPage={this.setCurrentPage}
+                    currentPage={this.state.currentPage}
+                    addLevel={this.props.addLevel}
+                    levels={this.state.levels}
+                    numPages={this.state.numPages}
+                  />
+                </div>
+              )}
+              {this.state.methodOfAddingLevel === 'Create New Level' && (
+                <CreateNewLevelInputs
+                  levelOptions={this.state.searchFields.levelOptions}
                   addLevel={this.props.addLevel}
-                  levels={this.state.levels}
                 />
-              </div>
-            )}
-            {this.state.methodOfAddingLevel === 'Create New Level' && (
-              <CreateNewLevelInputs
-                levelOptions={this.state.searchFields.levelOptions}
-                addLevel={this.props.addLevel}
-              />
-            )}
-          </div>
-          <div style={styles.rightColumn}>
+              )}
+            </div>
+          )}
+          {(!this.state.levels || !this.state.searchFields) && (
+            <FontAwesome icon="spinner" className="fa-spin" />
```

Comment:
Have you been able to verify that this is working? If not, this might be getting hit when I block one of the network requests as shown in my earlier screenshot. anyway, what I would recommend is to have levels and searchFields always contain valid values such an empty array of levels, and then use an explicit property to track whether a network request is in progress.